### PR TITLE
Fix topic and type name handling

### PR DESCRIPTION
**Pull request by pablogs9**

### CODE REVIEW

File: rmw_microxrcedds_c/src/rmw_microxrcedds_topic.c [hunk 1]
```diff
@@ -71,8 +71,19 @@ create_topic(
...
-  generate_type_name(message_type_support_callbacks, type_name, sizeof(type_name));
+  if (!generate_topic_name(topic_name, full_topic_name, sizeof(full_topic_name))) {
+    RMW_UROS_TRACE_MESSAGE("Error creating topic name");
+    rmw_uxrce_fini_topic_memory(custom_topic);
+    custom_topic = NULL;
+    goto fail;
+  }
+
+  if (!generate_type_name(message_type_support_callbacks, type_name, sizeof(type_name))) {
+    RMW_UROS_TRACE_MESSAGE("Error creating topic name");
```
Comment (pablogs9) [inline rmw_microxrcedds_c/src/rmw_microxrcedds_topic.c:82]:
```suggestion
    RMW_UROS_TRACE_MESSAGE("Error creating type name");
```

**Review by Acuadros95 — APPROVED** [review]

**Comment by Acuadros95** [comment]
LGTM

**Comment by pablogs9** [comment]
Thanks, @Acuadros95, let's wait for @gavanderhoorn feedback before merging.

**Comment by gavanderhoorn** [comment]
I haven't tested this yet, but I noticed the functions like `generate_topic_name(..)` now only return a `bool`.

`snprintf(..)` can fail in a nr of different ways, so it seems like returning a more fine-grained status/error code from `generate_topic_name(..)` et al. would make reporting the different failure modes from calling code also easier/possible.

Other places in Micro-ROS are very careful in how they handle and report errors (such as [the other parts of the ctor of rmw_client_t](https://github.com/micro-ROS/rmw_microxrcedds/blob/53df2d338fee96420a8658599f914651936019de/rmw_microxrcedds_c/src/rmw_client.c#L36-L53), a few lines up), so it would align with that approach.

What do you think?

**Comment by pablogs9** [comment]
`snprintf` will report that chars that should have been written are more than the buffer array, in that case, a false is returned. 

And the function will fail here for example https://github.com/micro-ROS/rmw_microxrcedds/pull/254/files#diff-fa1104ae182dc7153d689fedbab8020d187abaafeb7e9a6829346f040060302aR82

**Comment by gavanderhoorn** [comment]
I was just wondering whether we'd want to differentiate between 'buffer overrun' and 'other error', which according to [snprintf(3)](https://linux.die.net/man/3/snprintf) is possible.

**Comment by pablogs9** [comment]
Could you propose a user input that can end as "other error" in the internal usage of [snprintf(3)](https://linux.die.net/man/3/snprintf)?

**Comment by gavanderhoorn** [comment]
No, I can't come up with such a situation, but that's not really how I normally judge whether or not to handle return values the documentation tells me are possible :)

As I wrote, I was just curious whether you felt it would be beneficial to be able to distinguish between the different failure modes. More specific error messages have helped me in the past while debugging Micro-ROS issues (mostly in my personal use of the API, but still).

I'm OK with the proposed changes.

**Comment by gavanderhoorn** [comment]
Thanks for the quick turnaround btw. 👍

**Pull request merged**